src/main/scala/it/pagopa/interop/agreementprocess/error/AgreementProcessErrors.scala

```diff
@@ -99,5 +99,11 @@ object AgreementProcessErrors {
   final case class AttributeNotFound(attributeId: UUID)
       extends ComponentError("0022", s"Attribute ${attributeId.toString} not found")
 
-  case object InvalidAttributeStructure extends ComponentError("0023", "Invalid Attribute Structure")
+  final case class ConsumerWithNotValidEmail(agreementId: UUID, tenantId: UUID)
+      extends ComponentError(
+        "0023",
+        s"Agreement ${agreementId.toString} has a consumer tenant ${tenantId.toString} with no valid email"
+      )
+
+  case object InvalidAttributeStructure extends ComponentError("0024", "Invalid Attribute Structure")
```

Comment:
we should avoid to change error codes if not strictly required
```suggestion
  case object InvalidAttributeStructure extends ComponentError("0023", "Invalid Attribute Structure")
```